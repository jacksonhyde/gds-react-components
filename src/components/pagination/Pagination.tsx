import React, { HTMLProps } from 'react';
import classNames from 'classnames';
import { ArrowLeft, ArrowRight } from '../icons';

interface PaginationLinkProps extends HTMLProps<HTMLAnchorElement> {
  previous?: boolean;
  next?: boolean;
}

const PaginationLink: React.FC<PaginationLinkProps> = ({
  className,
  children,
  previous,
  next,
  ...rest
}) => (
  <li
    className={classNames(
      { 'govuk-pagination-item--previous': previous },
      { 'govuk-pagination-item--next': next },
    )}
  >
    <a
      className={classNames(
        'govuk-pagination__link',
        { 'govuk-pagination__link--prev': previous },
        { 'govuk-pagination__link--next': next },
        className,
      )}
      {...rest}
    >
      <span className="govuk-pagination__title">
        {previous ? 'Previous' : null}
        {next ? 'Next' : null}
      </span>
      <span className="govuk-u-visually-hidden">:</span>
      <span className="govuk-pagination__page">{children}</span>
      {previous ? <ArrowLeft /> : null}
      {next ? <ArrowRight /> : null}
    </a>
  </li>
);

interface Pagination extends React.FC<HTMLProps<HTMLDivElement>> {
  Link: React.FC<PaginationLinkProps>;
}

const Pagination: Pagination = ({ className, children, ...rest }) => (
  <nav className={classNames('govuk-pagination', className)} {...rest}>
    <ul className="govuk-list govuk-pagination__list">{children}</ul>
  </nav>
);

Pagination.defaultProps = {
  role: 'navigation',
  'aria-label': 'Pagination',
};

Pagination.Link = PaginationLink;

export default Pagination;
