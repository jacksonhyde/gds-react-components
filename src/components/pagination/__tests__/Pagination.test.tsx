import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '..';
import { ArrowLeft, ArrowRight } from '../../icons';

describe('Pagination', () => {
  it('matches snapshot', () => {
    const element = shallow(<Pagination />);
    expect(element).toMatchSnapshot('Pagination');
    element.unmount();
  });

  describe('Pagination.Link', () => {
    it('matches snapshot', () => {
      const element = shallow(<Pagination.Link />);
      expect(element).toMatchSnapshot('Pagination.Link');
      element.unmount();
    });

    it('renders previous elements', () => {
      const element = shallow(<Pagination.Link previous>PreviousText</Pagination.Link>);
      const paginationLink = element.find('.govuk-pagination__link');
      expect(element.hasClass('govuk-pagination-item--previous')).toBeTruthy();
      expect(element.hasClass('govuk-pagination-item--next')).toBeFalsy();
      expect(element.find('.govuk-pagination__title').text()).toBe('Previous');
      expect(element.contains(<ArrowLeft />)).toBeTruthy();
      expect(element.contains(<ArrowRight />)).toBeFalsy();
      expect(element.find('.govuk-pagination__page').text()).toBe('PreviousText');
      expect(paginationLink.hasClass('govuk-pagination__link--prev')).toBeTruthy();
      expect(paginationLink.hasClass('govuk-pagination__link--next')).toBeFalsy();
      element.unmount();
    });

    it('renders next elements', () => {
      const element = shallow(<Pagination.Link next>NextText</Pagination.Link>);
      const paginationLink = element.find('.govuk-pagination__link');
      expect(element.hasClass('govuk-pagination-item--previous')).toBeFalsy();
      expect(element.hasClass('govuk-pagination-item--next')).toBeTruthy();
      expect(element.find('.govuk-pagination__title').text()).toBe('Next');
      expect(element.contains(<ArrowLeft />)).toBeFalsy();
      expect(element.contains(<ArrowRight />)).toBeTruthy();
      expect(element.find('.govuk-pagination__page').text()).toBe('NextText');
      expect(paginationLink.hasClass('govuk-pagination__link--prev')).toBeFalsy();
      expect(paginationLink.hasClass('govuk-pagination__link--next')).toBeTruthy();
      element.unmount();
    });
  });
});
