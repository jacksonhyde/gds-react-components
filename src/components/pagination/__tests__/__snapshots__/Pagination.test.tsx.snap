// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Pagination.Link matches snapshot: Pagination.Link 1`] = `
<li
  className=""
>
  <a
    className="govuk-pagination__link"
  >
    <span
      className="govuk-pagination__title"
    />
    <span
      className="govuk-u-visually-hidden"
    >
      :
    </span>
    <span
      className="govuk-pagination__page"
    />
  </a>
</li>
`;

exports[`Pagination matches snapshot: Pagination 1`] = `
<nav
  aria-label="Pagination"
  className="govuk-pagination"
  role="navigation"
>
  <ul
    className="govuk-list govuk-pagination__list"
  />
</nav>
`;
